nc/source/ecs/EcsApiImpl.cpp

```diff
@@ -14,20 +14,10 @@ namespace nc
         }
     }
 
-    EntityHandle CreateEntity()
+    EntityHandle CreateEntity(EntityInfo info)
     {
-        return CreateEntity(Vector3::Zero(), Quaternion::Identity(), Vector3::One(), internal::DefaultEntityTag);
-    }
-
-    EntityHandle CreateEntity(std::string tag)
-    {
-        return CreateEntity(Vector3::Zero(), Quaternion::Identity(), Vector3::One(), std::move(tag));
-    }
-
-    EntityHandle CreateEntity(Vector3 pos, Quaternion rot, Vector3 scale, std::string tag)
-    {
-        IF_THROW(scale == Vector3::Zero(), "CreateEntity - scale cannot be Vector3::Zero");
-        return internal::g_impl->CreateEntity(pos, rot, scale, std::move(tag));
+        IF_THROW(info.scale == Vector3::Zero(), "CreateEntity - scale cannot be Vector3::Zero");
```

Comment:
Not sure about best practices with structs, but I would like this type checking to be done at struct creation time, so we aren't passing around potentially invalid structs. Maybe a CTOR or explicit setter?